crates/fuel-core/src/coins_query.rs

```diff
@@ -277,12 +287,12 @@ pub async fn select_coins_to_spend(
     let (selected_big_coins_total, selected_big_coins) =
         big_coins(big_coins_stream, adjusted_total, max, exclude).await?;
 
-    if selected_big_coins_total < total {
+    if selected_big_coins_total < total && !allow_partial {
```

Comment:
I think if `selected_big_coins_total` is zero we still should return an error

Reply:
Done and also modified in the other function